crates/astria-bridge-signer/src/verifier.rs

```diff
@@ -0,0 +1,312 @@
+use std::{
+    str::FromStr as _,
+    sync::Arc,
+};
+
+use astria_core::{
+    generated::astria::protocol::transaction::v1::TransactionBody as RawTransactionBody,
+    primitive::v1::{
+        asset::Denom,
+        Address,
+    },
+    protocol::transaction::v1::{
+        action::{
+            BridgeUnlock,
+            Ics20Withdrawal,
+        },
+        Action,
+        TransactionBody,
+    },
+    Protobuf as _,
+};
+use astria_eyre::eyre::{
+    self,
+    bail,
+    ensure,
+    eyre,
+    OptionExt as _,
+    WrapErr as _,
+};
+use ethers::{
+    providers::{
+        Http,
+        Middleware,
+        Provider,
+    },
+    types::H256,
+    utils::hex,
+};
+use prost::Message as _;
+
+enum GetterConfiguration<'a> {
+    SequencerWithdrawal,
+    Ics20Withdrawal(&'a Denom),
+}
+
+pub struct Verifier {
+    provider: Arc<Provider<Http>>,
+}
+
+impl Verifier {
+    /// Creates a new `Verifier` with the given rollup RPC endpoint.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the provider cannot be created.
+    pub fn new(rollup_rpc_endpoint: String) -> eyre::Result<Self> {
+        let provider = Provider::<Http>::try_from(rollup_rpc_endpoint)
+            .wrap_err("failed to create provider")?;
```

Comment:
```suggestion
                .wrap_err("failed to create rollup client")?;
```
Or maybe `ethereum client`, but `provider` is meaningless and doesn't carry any information.